src/core/control/tools/EditSelection.cpp

```diff
@@ -312,6 +312,22 @@ auto EditSelection::setFont(XojFont& font) -> UndoAction* { return this->content
  */
 void EditSelection::fillUndoItem(DeleteUndoAction* undo) { this->contents->fillUndoItem(undo); }
 
+/**
+ * Computes the reflection with respect to either the horizontal or vertical axis.
+ *
+ */
+auto EditSelection::reflectSelection(bool x_axis) -> UndoAction* {
+    PageRef page = this->view->getPage();
+    Layer* layer = page->getSelectedLayer();
+    this->rotation = snappingHandler.snapAngle(this->rotation, false);
+    this->sourcePage = page;
+    this->sourceLayer = layer;
+    this->contents->updateContent(this->getRect(), this->snappedBounds, this->rotation, this->aspectRatio, layer, page,
+                                  this->view, this->undo, this->mouseDownType);
+    finalizeSelection();
```

Comment:
`finalizeSelection` should not be called here, otherwise the elements will be added to the page.

Reply:
Done in `c4bf75`